lib/house.rb

```diff
@@ -0,0 +1,65 @@
+class House
+  attr_reader :address, :rooms
+
+  def initialize(price_paramater, address_parameter)
+    @price = price_paramater
+    @address = address_parameter
+    @rooms = []
+  end
+
+  def price
+    @price.gsub("$", '').to_i
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    price > 500000 ? true : false
+  end
+
+  def rooms_from_category(category)
+    @rooms.select {|room| room.category == category}
+  end
+
+  def area
+    total_area = 0
+    @rooms.each {|room| total_area += room.area}
+    total_area
+  end
+
+  def details
+    details = {"price" => price, "address" => @address}
+    details
+  end
+
+  def price_per_square_foot
+    total_per_square_foot = 0
+    if area != 0
+      total_per_square_foot = (price / area.to_f).round(2)
+    end
+    total_per_square_foot
```

Comment:
good job accounting for the 0 here, although if you do that, i would provide an alternative option if area does == 0, as a matter of best practice.  Nice job using the return value of another method as a helper here.